internal/cache/file/downloader/parallel_downloads_job.go

```diff
@@ -60,7 +61,14 @@ func (job *Job) downloadRange(ctx context.Context, dstWriter io.Writer, start, e
 
 	monitor.CaptureGCSReadMetrics(ctx, util.Parallel, end-start)
 
-	_, err = io.CopyN(dstWriter, newReader, end-start)
+	_, err = cacheutil.CopyUsingMemoryAlignedBuffer(ctx, newReader, dstWriter, end-start,
+		job.fileCacheConfig.WriteBufferSize)
+	// If context is canceled while reading/writing in CopyUsingMemoryAlignedBuffer
+	// then it returns invalid argument error, and we need to report that error as
+	// context cancelled.
+	if errors.Is(ctx.Err(), context.Canceled) {
```

Comment:
I didnt follow this. You mentioned in the comment that it will return invalid argument error and checking if it is context.Cancelled error?

Reply:
Will update the comment to be more clear but the intention is that if the context is canceled then we can return context canceled error. (we are doing this because the copy function can return other error - invalid argument)